web/vars_native.go

```diff
@@ -0,0 +1,45 @@
+// +build go1.7
+
+package web
+
+import (
+	"context"
+	"net/http"
+)
+
+// Vars is a helper function for accessing route
+// parameters from any server.Router implementation. This is the equivalent
+// of using `mux.Vars(r)` with the Gorilla mux.Router.
+func Vars(r *http.Request) map[string]string {
+	// vars doesnt exist yet, return empty map
+	rawVars := r.Context().Value(varsKey)
+	if rawVars == nil {
+		return map[string]string{}
+	}
+
+	// for some reason, vars is wrong type, return empty map
+	vars, ok := rawVars.(map[string]string)
+	if !ok {
+		return map[string]string{}
```

Comment:
Since the gorilla returns nil when it's not defined, should this return nil as well?

It almost makes no difference for the caller (I wouldn't expect someone to modify the map returned by this function), but it could make the code here way simpler, something like:

```
rawVars := r.Context().Value(varsKey)
vars, _ := rawVars.(map[string]string)
return vars
```